lib/puppet/parser/scope.rb

```diff
@@ -1041,7 +1041,11 @@ def method_missing(method, *args, &block)
     if respond_to? method
       send(method, *args)
     else
-      raise Puppet::DevError, _("Function %{name} not defined despite being loaded!") % { name: name }
+      if caller.any? { |line| line =~ /runtime3_support.rb/ }
```

Comment:
`line.include?('runtime3_support.rb')` could be faster but I'm not sure

Reply:
Can we determine this without going through the entire stack trace? It's a very expensive operation in JRuby.

Reply:
I'll check to see if this can be determined without searching in the stack trace.